### ISSUE TITLE
Melee combat

### ISSUE THREAD

**Issue by Dyhr** [RE-SS3D/Client#45]
You should be able to swing at object to hit another character or item

**Comment by JJRcop** [comment]
Discussion: Should it be click on target, or a swing animation with a hit check?

**Comment by Dyhr** [comment]
That would depend on how intents work. Traditionally, swinging happens when you click on a target.
Alternatively if we change intents a bit then I think it would make sense that help intent never swings your weapon and harm mode always swings your weapon.

**Comment by Dyhr** [comment]
Swing animation with hit check would otherwise unintentionally hit your surroundings all the time

**Comment by AWalton** [comment]
I think it should always do "swing animation with a hit check" since you toolbox people even on Help intent. Exceptions to this would be stuff that has a possible Help intent alternate action like syringes (inject/draw), mop (clean), etc.

**Comment by Dyhr** [comment]
> 
> 
> I think it should always do "swing animation with a hit check" since you toolbox people even on Help intent. Exceptions to this would be stuff that has a possible Help intent alternate action like syringes (inject/draw), mop (clean), etc.
> 
> If you don't mean to be toolboxing your surroundings then you should be clicking with your empty other hand.

I would much rather stay consistent with SS13, and only swing when you click a target, rather than making it even less user friendly.

**Comment by AWalton** [comment]
Having to click perfectly on the small sprite of an enemy whilst you're both moving at high speeds is what has made SS13 melee combat so difficult and user-unfriendly. I always thought this was a limitation of being 2D rather than it being an essential stylistic feature, because each "swing" would necessitate doing hitscans on up to 2 tiles, which would have made melee too easy in 2D. 

Beep's original videos in contrast had it that you could melee your surroundings with items also https://youtu.be/zVMNHL_Rwl0. And when you compare it against combat of games with similar art styles (Gang Beasts, Breath of the Wild) or 3D RPGs with many systems including complex melee and ranged combat (Witcher 3, Elder Scrolls: Skyrim, Mass Effect 3) it always does hitscans for melee too. 

It wouldn't be user-friendly to be facing someone all ready to swing your improvised weapon but literally nothing happens just because it's on the wrong intent.

**Comment by Dyhr** [comment]
Yeah, I agree. But I would love to change help intent so it doesn't do any harm at all. It seems like someone forgot about intents when they implemented attacking in SS13 and just implemented it as a default action.

We probably need a definition for user friendly, but in my view, we should try and prevent accidental destructive actions. Just like when a system ask you if you are sure that you want to permanently delete that file, it's nice to have a safety to rely on.

**Comment by AWalton** [comment]
Makes sense, though it needs to be easier to change intents if so e.g. hold SHIFT to perform actions with a HELP intent. Perhaps a customisable hotkeys milestone in the near future.
